src/main/java/ui/components/pickers/MilestonePickerState.java

```diff
@@ -105,13 +105,15 @@ private void toggleMilestone(PickerMilestone milestone) {
      */
     private void populateBestMatchingMilestones(String querySentence) {
         bestMatchingMilestones.clear();
-        addMatchingMilestones(bestMatchingMilestones, getRemainingBestMatchesToLimit(), allMilestones);
+        addMatchingMilestones(bestMatchingMilestones, getRemainingBestMatchesToLimit(),
+                                                      getSelectableMilestones(allMilestones));
         if (isMilestonesSizeBelowLimit(bestMatchingMilestones, BEST_MATCHING_LIMIT)) {
             addPartiallyMatchingMilestones(bestMatchingMilestones, getRemainingBestMatchesToLimit(), querySentence,
-                                            allMilestones);
+                                           getSelectableMilestones(allMilestones));
         }
         if (isMilestonesSizeBelowLimit(bestMatchingMilestones, BEST_MATCHING_LIMIT)) {
-            addLikelyUnmatchedMilestones(bestMatchingMilestones, getRemainingBestMatchesToLimit(), allMilestones);
+            addLikelyUnmatchedMilestones(bestMatchingMilestones, getRemainingBestMatchesToLimit(),
+                                         getSelectableMilestones(allMilestones));
```

Comment:
Use the variable `selectableMilestones`? I mean, you don't want to do the operations again and again. Also, the variable name already represents the high-level logic.